### PR TITLE
Add options to be able to configure telegraf as a worker kafka to influxDb

### PR DESCRIPTION
**Pull request by lolemetayer**
Issue appcelerator/amp-project#115
- Add options to enable/disable inputs (cpu, mem, etc.). 
- Add a Kafka input to act as a worker. 
- Prefix by OUTPUT env vars for kafka output (and distinguish Output and Input)
- Add option to increase InfluxDB timeout (could help on slow dev environment)

By default:
- InfluxDB output is ~~disabled (updated)~~ enabled (as previously) [edited]
- Kafka output is ~~enabled (updated)~~ disabled (as previously) [edited]
- Docker input is enabled (as previously)
- cpu, mem, ... is disabled (updated)

### CODE REVIEW

**Comment by ndegory** [comment]
Please keep the InfluxDB output enabled by default, that what we would expect from telegraf

**Comment by ndegory** [comment]
Please keep the Kafka output disabled by default, that what we would expect from telegraf

**Comment by lolemetayer** [comment]
Like a standard telegraf, standard output  is InfluxDb and Kafka is disabled.

File: Dockerfile [hunk 1]
```diff
@@ -26,6 +26,16 @@ ENV INTERVAL 10s
...
+ENV INPUT_KAFKA_ENABLED         false
+ENV INPUT_CPU_ENABLED           false
+ENV INPUT_DISK_ENABLED          false
+ENV INPUT_DISKIO_ENABLED        false
+ENV INPUT_KERNEL_ENABLED        false
+ENV INPUT_MEM_ENABLED           false
+ENV INPUT_PROCESS_ENABLED       false
+ENV INPUT_SWAP_ENABLED          false
+ENV INPUT_SYSTEM_ENABLED        false
+ENV INPUT_DCOKER_ENABLED        true
```
Comment (ndegory) [inline Dockerfile]: dyslexics of the world, untie!

File: telegraf.conf.tpl [hunk 2]
```diff
@@ -184,52 +188,75 @@
...
+{% if INPUT_SYSTEM_ENABLED == "true" %}
 [[inputs.system]]
   # no configuration
-
+{% else %}
+  # System input is disabled
+{% endif %}
 
 # Read metrics about docker containers
+{% if INPUT_DCOKER_ENABLED == "true" %}
```
Comment (ndegory) [inline telegraf.conf.tpl]: same here

File: telegraf.conf.tpl [hunk 3]
```diff
@@ -250,3 +277,26 @@
 # [[inputs.netstat]]
 #   # no configuration
 
+# Read metrics from Kafka topic(s)
+{% if INPUT_KAFKA_ENABLED == "true" %}
+[[inputs.kafka_consumer]]
+  ## topic(s) to consume
+  topics = ["telegraf"]
```
Comment (ndegory) [inline telegraf.conf.tpl]: I think we need a INPUT_KAFKA_TOPIC env var, default value "telegraf"

File: telegraf.conf.tpl [hunk 4]
```diff
@@ -250,3 +277,26 @@
...
+  ## an array of Zookeeper connection strings
+  zookeeper_peers = ["zookeeper:2181"]
```
Comment (ndegory) [inline telegraf.conf.tpl]: not sure how it is used, but clearly we need a INPUT_KAFKA_ZOOKEEPER_PEER env var

File: telegraf.conf.tpl [hunk 5]
```diff
@@ -250,3 +277,26 @@
...
+  metric_buffer = 100000
+  ## Offset (must be either "oldest" or "newest")
+  offset = "oldest"
+
+  ## Data format to consume.
+
+  ## Each data format has it's own unique set of configuration options, read
+  ## more about them here:
+  ## https://github.com/influxdata/telegraf/blob/master/docs/DATA_FORMATS_INPUT.md
+  data_format = "influx"
```
Comment (ndegory) [inline telegraf.conf.tpl]: Let's use the existing KAFKA_DATA_FORMAT env var

**Comment by ndegory** [comment]
changes are requested, as commented inline

**Comment by lolemetayer** [comment]
All comments fixed in da78e08

**Comment by lolemetayer** [comment]
:ok_hand: Ok for me. Pull request to proceed.

File: README.md [hunk 6]
```diff
@@ -29,13 +29,27 @@ docker run -t -v /var/run/docker.sock:/var/run/docker.sock:ro -v /var/run/utmp:/
...
+- INFLUXDB_TIMEOUT - InfluxDB timetout (in seconds), defaults to 5
 - CLOUDWATCH_REGION - Amazon region, defaults to us-east-1
 - CLOUDWATCH_NAMESPACE - Namespace, defaults to InfluxData/Telegraf
-- KAFKA_BROKER_URL - Kafka broker URL, defaults to localhost:9092
-- KAFKA_TOPIC - Kafka topic, defaults to telegraf
+- OUTPUT_KAFKA_BROKER_URL - Kafka broker URL in output, defaults to localhost:9092
+- OUTPUT_KAFKA_TOPIC - Kafka topic on which to write, defaults to telegraf
+- INPUT_KAFKA_BROKER_URL - Kafka broker URL in input, defaults to localhost:9092
+- INPUT_KAFKA_TOPIC - Kafka topic on which to read, defaults to telegraf
+- INPUT_KAFKA_ZOOKEEPER_PEER - Zookeeper peers used by Kafka in input, default to zookeeper:2181
```
Comment (ndegory) [inline README.md]: defaultS

File: telegraf.conf.tpl [hunk 7]
```diff
@@ -113,9 +113,9 @@
 {% if OUTPUT_KAFKA_ENABLED == "true" %}
 [[outputs.kafka]]
   ## URLs of kafka brokers
-  brokers = ["{{ KAFKA_BROKER_URL | default("localhost:9092") }}"]
+  brokers = ["{{ OUTPUT_KAFKA_BROKER_URL | default("localhost:9092") }}"]
   ## Kafka topic for producer messages
-  topic = "{{ KAFKA_TOPIC | default("telegraf") }}"
+  topics = ["{{ OUTPUT_KAFKA_TOPIC | default("telegraf") }}"]
```
Comment (ndegory) [inline telegraf.conf.tpl]: topics or topic?

File: telegraf.conf.tpl [hunk 8]
```diff
@@ -250,3 +277,26 @@
 # [[inputs.netstat]]
 #   # no configuration
 
+# Read metrics from Kafka topic(s)
+{% if INPUT_KAFKA_ENABLED == "true" %}
+[[inputs.kafka_consumer]]
+  ## topic(s) to consume
+  topic = "{{ INPUT_KAFKA_TOPIC | default("telegraf") }}"
```
Comment (ndegory) [inline telegraf.conf.tpl]: topic or topics?

**Comment by ndegory** [comment]
LGTM

**Pull request merged**